infra/event/EventDispatcherWithWeakPtr.cpp

```diff
@@ -46,6 +46,19 @@ namespace infra
         {}
     }
 
+    void EventDispatcherWithWeakPtrWorker::ExecuteUntil(const infra::Function<bool()>& predicate)
+    {
+        while (!predicate())
+        {
+            ExecuteAllActions();
+
+            if (predicate())
+                break;
```

Comment:
```suggestion
                return;
```

Reply:
I actually prefer break over return. Post condition of the while-loop is that predicate() hold. Post condition has been found to hold, so we break from the loop.